app/controllers/patients_controller.rb

```diff
@@ -542,6 +548,105 @@ def update_monitoring_fields(patient, params, household, propagation)
     patient.monitoring_history_edit(history_data, diff_state)
   end
 
+  # Update the patient's follow-up flag fields
+  def update_follow_up_flag
+    redirect_to(root_url) && return unless current_user.can_edit_patient? && !current_user.role?(Roles::ENROLLER)
+
+    patient = current_user.get_patient(params.permit(:id)[:id])
+    redirect_to(root_url) && return if patient.nil?
+
+    update_follow_up_flag_fields(patient, params)
+  end
+
+  # Make updates to follow-up flag reason and/or note and create corresponding History items.
+  #
+  # patient - The Patient to update.
+  # params - The request params.
+  def update_follow_up_flag_fields(patient, params)
+    clear_flag = params.permit(:clear_flag)[:clear_flag]
+    history_data = {}
+    if clear_flag
+      clear_flag_reason = params.permit(:clear_flag_reason)[:clear_flag_reason]
+      clear_follow_up_flag(patient, clear_flag_reason)
+    else
+      follow_up_reason_before = patient.follow_up_reason
+      follow_up_note_before = patient.follow_up_note
+      follow_up_reason = params.permit(:follow_up_reason)[:follow_up_reason]
+      follow_up_note = params.permit(:follow_up_note)[:follow_up_note]
+
+      # Apply and save updates to the db
+      patient.update!(follow_up_reason: follow_up_reason, follow_up_note: follow_up_note)
+
+      # Handle creating history items based on the updates
+      history_data = {
+        created_by: current_user.email,
+        patient: patient,
+        follow_up_reason: follow_up_reason,
+        follow_up_note: follow_up_note,
+        follow_up_reason_before: follow_up_reason_before,
+        follow_up_note_before: follow_up_note_before
+      }
+
+      History.follow_up_flag_edit(history_data)
+    end
+
+    # If not applying to household, return
+    apply_to_household_ids = params.permit(apply_to_household_ids: [])[:apply_to_household_ids]
+    return unless params.permit(:apply_to_household)[:apply_to_household] && !apply_to_household_ids.nil?
```

Comment:
minor style suggestion

```suggestion
    return unless params.permit(:apply_to_household)[:apply_to_household] && apply_to_household_ids.present?
```